pinot-core/src/main/java/org/apache/pinot/core/transport/ServerChannels.java

```diff
@@ -167,7 +167,10 @@ protected void initChannel(SocketChannel ch) {
               // NOTE: data table de-serialization happens inside this handler
               // Revisit if this becomes a bottleneck
               ch.pipeline().addLast(
-                  ChannelHandlerFactory.getDataTableHandler(_queryRouter, _serverRoutingInstance, _brokerMetrics));
+                  ChannelHandlerFactory.getDirectOOMHandler(_queryRouter, _serverRoutingInstance, _serverToChannelMap)
```

Comment:
We should add the new handler before the comment note

Reply:
Done